digests.go

```diff
@@ -13,23 +13,23 @@ type Digester interface {
 	Digest(io.Reader) (digest.Digest, error)
 }
 
-// ContentProvider produces a read stream for a given digest
+// ContentProvider provides content for a digest.
 type ContentProvider interface {
+	// Reader returns a stream of the content for
+	// the provided digest.
 	Reader(digest.Digest) (io.ReadCloser, error)
+
+	// CopyTo copies the content from the provided digest
+	// to the destination file path.
+	CopyTo(string, digest.Digest) error
```

Comment:
Let me try keeping the interface the same and asserting for `*os.File` (or maybe has `Fd() uintptr`). I can probably just hide the use of `CopyFileRange` down in the apply rather than leaving it up the implementer to attempt. (It is a bit low level). The CopyFileRange interface does make it a bit trickier to use because of its requirement to have an offset, I can either assume an offset at the beginning in that case or read for the current offset.